tmt/steps/prepare/feature.py

```diff
@@ -0,0 +1,268 @@
+            if os_release.get('VERSION_ID', '').startswith('8'):
+                return Distro.RHEL_8
+            if os_release.get('VERSION_ID', '').startswith('7'):
+                return Distro.RHEL_7
+
+        if os_release.get('NAME') == 'CentOS Linux' and os_release.get(
+                'VERSION_ID', '').startswith('7'):
+            return Distro.CENTOS_7
+
+        if os_release.get('NAME') == 'CentOS Stream':
+            if os_release.get('VERSION_ID', '').startswith('8'):
+                return Distro.CENTOS_STREAM_8
+            if os_release.get('VERSION_ID', '').startswith('9'):
+                return Distro.CENTOS_STREAM_9
+
+        return None
+
+
+class ToggleableFeature(Feature):
+    def enable(self, guest: Guest, logger: tmt.log.Logger) -> None:
+        raise NotImplementedError
+
+    def disable(self, guest: Guest, logger: tmt.log.Logger) -> None:
+        raise NotImplementedError
+
+
+class CRB(ToggleableFeature):
+    KEY = 'crb'
+    # TBD
+
+
+class FIPS(ToggleableFeature):
+    KEY = 'fips'
+    # TBD
+
+
+FEDORA_REPO = 'powertools'
+FEDORA_PACKAGES = ['epel-release', 'epel-next-release']
+
+CENTOS_7_PACKAGES = ['epel-release']
+
+CENTOS_STREAM_9_REPO = 'crb'
+
+RHEL_7_REPO = 'rhel-*-optional-rpms rhel-*-extras-rpms rhel-ha-for-rhel-*-server-rpms'
+RHEL_7_PACKAGES = ['https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm']
+
+RHEL_8_REPO = 'codeready-builder-for-rhel-8-$(arch)-rpms'
+RHEL_8_PACKAGES = ['https://dl.fedoraproject.org/pub/epel/epel-release-latest-8.noarch.rpm']
+
+RHEL_9_REPO = 'codeready-builder-for-rhel-9-$(arch)-rpms'
+RHEL_9_PACKAGES = ['https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm']
+
+
+class EPEL(ToggleableFeature):
+    KEY = 'epel'
+
+    def enable(self, guest: Guest, logger: tmt.log.Logger) -> None:
+        guest_distro = self.get_guest_distro(guest=guest, logger=logger)
+        if guest_distro is None:
+            raise tmt.utils.PrepareError('The distro of the guest is unsupported.')
+
+        sudo = 'sudo' if self.guest.facts.is_superuser is False else ''
+
+        if guest_distro == Distro.FEDORA:
+            self.info('Nothing to do on Fedora for EPEL')
+        elif guest_distro == Distro.CENTOS_7:
+            # yum install epel-release
+            self.info('Enable EPEL on CentOS 7')
+            shscript1 = ShellScript(f'{sudo} yum -y install {" ".join(CENTOS_7_PACKAGES)}')
+            self.guest.execute(shscript1, silent=True)
+        elif guest_distro == Distro.CENTOS_STREAM_8:
+            # dnf config-manager --set-enabled powertools
+            # dnf -y install epel-release epel-next-release
+            self.info('Enable EPEL on CentOS Stream 8')
+            shscript1 = ShellScript(f'{sudo} dnf config-manager --set-enabled {FEDORA_REPO}')
+            shscript2 = ShellScript(f'{sudo} dnf -y install {" ".join(FEDORA_PACKAGES)}')
+            self.guest.execute(shscript1 & shscript2, silent=True)
```

Comment:
`<nitpicking>` & `<personal pov>` :)

I for one am not very fond of having variables like `shscript1` to hold values that can be used directly, IMHO it makes code harder to read. I need to look back at line 129 to recall what exactly is being executed. With a bit of indentation, we could have something

```python
shscript1 = ShellScript(f'{sudo} dnf config-manager --set-enabled {FEDORA_REPO}')
shscript2 = ShellScript(f'{sudo} dnf -y install {" ".join(FEDORA_PACKAGES)}')
self.guest.execute(shscript1 & shscript2, silent=True)

# vs

self.guest.execute(
      ShellScript(f'{sudo} dnf config-manager --set-enabled {FEDORA_REPO}')
    & ShellScript(f'{sudo} dnf -y install {" ".join(FEDORA_PACKAGES)}'),
    silent=True)
```

Reply:
Updated, thanks for the suggestion :-)